### PR TITLE
TF2 Upgrader: add support of ipynb files

### PR DESCRIPTION
**Pull request by lc0**
As discussed @martinwicke, since locally it was tested in isolated environment, I would be more than happy to run the required tests

Potentially I'd follow with updating readme and other docs related things

### CODE REVIEW

**Comment by lc0** [comment]
I think once, https://github.com/tensorflow/tensorflow/pull/25653 is in, I can rebase this one

File: tensorflow/tools/compatibility/tf_upgrade_v2_main.py [hunk 1]
```diff
@@ -22,15 +22,17 @@
...
 from tensorflow.tools.compatibility import ast_edits
 from tensorflow.tools.compatibility import tf_upgrade_v2
+from tensorflow.tools.compatibility import ipynb
 
 
 def main():
   parser = argparse.ArgumentParser(
       formatter_class=argparse.RawDescriptionHelpFormatter,
-      description="""Convert a TensorFlow Python file to 2.0
+      description="""Convert a TensorFlow Python file from 1.* to 2.0
```
Comment (martinwicke) [inline tensorflow/tools/compatibility/tf_upgrade_v2_main.py]: I prefer 1.x

File: tensorflow/tools/compatibility/tf_upgrade_v2_main.py [hunk 2]
```diff
@@ -85,7 +87,7 @@ def main():
       raise ValueError(
           "--outfile=<output file> argument is required when converting a "
           "single file.")
-    files_processed, report_text, errors = upgrade.process_file(
+    files_processed, report_text, errors = ipynb.process_file(
```
Comment (martinwicke) [inline tensorflow/tools/compatibility/tf_upgrade_v2_main.py]: I'd prefer the filename selection logic to live here.

Comment (martinwicke) [reply tensorflow/tools/compatibility/tf_upgrade_v2_main.py]: Also, look at line 99, we should make sure that convert tree also understands ipynb files.

File: tensorflow/tools/compatibility/ipynb.py [hunk 3]
```diff
@@ -0,0 +1,138 @@
...
+
+def process_file(in_filename, out_filename):
+  """The function where we inject the support for ipynb upgrade."""
+  upgrader = ast_edits.ASTCodeUpgrader(tf_upgrade_v2.TFAPIChangeSpec())
+
+  if in_filename.endswith('.py'):
+    files_processed, report_text, errors = \
+      upgrader.process_file(in_filename, out_filename)
+
+  elif in_filename.endswith('.ipynb'):
```
Comment (martinwicke) [inline tensorflow/tools/compatibility/ipynb.py]: Can you move the dispatch logic to the main function (and possibly maybe even process_tree)

**Review by martinwicke — CHANGES_REQUESTED** [review]
This is super-awesome!

File: tensorflow/tools/compatibility/tf_upgrade_v2_main.py (hunk 1 above)
Comment (lc0) [reply tensorflow/tools/compatibility/tf_upgrade_v2_main.py]: funny, since I started as `1.x`, but in a header, it was already written as `1.*`

**Comment by lc0** [comment]
@martinwicke Thanks for review, I've updated accordingly. 
And with tree part, I would add it as a second PR potentially.

**Review by martinwicke — DISMISSED** [review]
LG, doing the tree stuff in a separate PR makes sense.

**Comment by lc0** [comment]
@martinwicke I fixed linter and other build issues. Please let me know if anything additional is missing.

**Pull request merged**